### PR TITLE
fix(grid-container): children proptype definition (FE-3222)

### PR DESCRIPTION
**Pull request by ianforster-sage**
fixes: #3272

### Proposed behaviour
This fix clears the console warning when a single GridItem is passed to the children prop by correctly defining the children proptype as GridItem or Array of GridItem. 

### Current behaviour
Single GridItem as a child of GridContainer throws this console warning
`Warning: Failed prop type: Invalid prop `children` supplied to `GridContainer`.`

### Checklist

- [x] Commits follow our style guide
~- [ ] Screenshots are included in the PR if useful~
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
~- [ ] Cypress automation tests added or updated if required~
~- [ ] Storybook added or updated if required~
~- [ ] Typescript `d.ts` file added or updated if required~
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
Use the codesandbox generated for issue 3272 below, edit it to pass either a single GridItem or multiple GridItems there should no longer be a console warning.

### CODE REVIEW

**Comment by grabkowski** [comment]
you could still place a component and a grid item without a warning:
https://codesandbox.io/s/young-field-coc6v

File: src/components/grid/grid-container/grid-container.component.js [hunk 1]
```diff
@@ -32,8 +32,12 @@ const GridContainer = (props) => {
...
 GridContainer.propTypes = {
   /** Defines the Components to be rendered within the GridContainer. Requires a GridItem */
-  children: PropTypes.oneOfType([GridItem, PropTypes.arrayOf(GridItem)])
-    .isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.shape({
+      type: PropTypes.oneOf([GridItem]),
+    }),
+    PropTypes.arrayOf(GridItem),
+  ]).isRequired,
```
Comment (grabkowski) [inline src/components/grid/grid-container/grid-container.component.js:40]: we need the same check on the array:

```
const gridItemType =PropTypes.shape({
    type: PropTypes.oneOf([GridItem])
})
.
.
.
children: PropTypes.oneOfType([
  gridItemType,
  PropTypes.arrayOf(gridItemType),
]).isRequired,
```

**Review by grabkowski — DISMISSED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Review by sianford — APPROVED** [review]

**Comment by sianford** [comment]
Tested in codesandbox https://codesandbox.io/s/pensive-dream-gieik?file=/src/index.js in Safari, Chrome, Firefox and Edge Chromium. Checked there were no console errors when there were 0, 1, 2 or 3 grid items. Amended the default story of Grid and removed two grid items leaving just one. There was no console error. All Cypress and Chromatic tests have passed. No changes required to any automation tests. Testing passed ✅

**Pull request merged**